app/scripts/models/reliers/oauth.js

```diff
@@ -204,6 +207,41 @@ define(function (require, exports, module) {
     },
 
     /**
+     * Check if the relier wants access to the account encryption keys.
+     *
+     * @returns {Boolean}
+     */
+    wantsKeys () {
+      return !! (this.has('keysJwk') && this._config.scopedKeysEnabled && this._validateKeyScope());
```

Comment:
@rfk so to address concerns here:

1. when we are at `wantsKeys` we already made the request for client details
1. we do not use the redirect_uri from params for stuff, we rely on client info from the server
1. i added this extra validation here `_validateKeyScope` that should catch `scope` and `redirectUri` mismatch. 
the oauth-server would catch this case already by matching client ids and requested scopes.

Let me know your thoughts and if this is sufficient!

Reply:
> we do not use the redirect_uri from params for stuff, we rely on client info from the server

Do we error out if there's a mismatch between the one from params and the one from the server?

Reply:
> Do we error out if there's a mismatch between the one from params and the one from the server?

No

Reply:
@rfk I think the uri param is just largely ignored. it is validated as a uri though

Reply:
Looks like we check it on the oauth-server here anyway: https://github.com/mozilla/fxa-oauth-server/blob/master/lib/routes/authorization.js#L239 👍